### ISSUE TITLE
Cannot restore interface to integrators

### ISSUE THREAD

**Issue by sroet** [choderalab/openmmtools#348]
This is following on openpathsampling/openpathsampling#767

Created a minimal failing example:
```python
from simtk.openmm import XmlSerializer
from openmmtools.utils import RestorableOpenMMObject
import openmmtools.integrators

with open('integrator_xml.txt', 'r') as f:
    xml = f.read()

integrator = XmlSerializer.deserialize(xml)
succes = RestorableOpenMMObject.restore_interface(integrator)
```
This fails on a clean conda install of just `openmmtools` (py 3.6) also failed in py 2.7. The xml can be restored in `openmmtools=0.14` (with appropriate modifications for importing the restoring object)

[integrator_xml.txt](https://github.com/choderalab/openmmtools/files/1916145/integrator_xml.txt)

**Comment by andrrizzi** [comment]
Hmm. This is weird. I've tried with `openmmtools 0.14`, but it fails to restore the integrator too. The only difference is that `openmmtools 0.14` just returns `False` when that happens while `0.15` now raises an exception.

The only hypothesis I have in mind right now is that something went wrong when the serialization was _created_ (i.e. the stored hash is somehow incorrect). I haven't changed the hashing algorithm since I created the system. This is the call:
```
float(zlib.adler32(openmm_class.__name__.encode()))
```
I doubt the result of this call depends on the OS it is executed, but I'll double check.

Do you know which code and which Python version created this integrator?

cc: @dwhswenson

**Comment by dwhswenson** [comment]
> Do you know which code and which Python version created this integrator?

Definitely Python 2.7, but the integrator is *old* (with OPS, we try to create objects once and then re-save them in other files -- that way we're guaranteed to use the same settings).

I don't know if we'll be able to exactly trace the provenance, but it is likely that @sroet's integrator goes back to `openmmtools 0.9` (and #210).

**Comment by andrrizzi** [comment]
Auch! Now I remember I did change the hashing algorithm between in 0.11. In the first implementation, I was using this to create the hash:
```python
float(hash(integrator_class.__name__))
```
but I had to change it because the built-in `hash()` function is seeded by `PYTHONHASHSEED`, which made the hash inconsistent across different processes. Initially I designed the system only to make it possible to retrieve the correct integrator after a copy or from `Context.getIntegrator()` so it wasn't a problem there.

If you already know the class of your integrator, this hack should work
```python
integrator = XmlSerializer.deserialize(xml)
integrator.__class__ = openmmtools.integrators.VVVRIntegrator
```
but I'm not sure if it's possible to make the current OpenMMTools backwards compatible with disk serialization before `openmmtools 0.11`.
